src/components/PeopleTable.tsx

```diff
@@ -3,3 +1,105 @@
+import React from 'react';
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
+import { PersonLink } from './PersonLink';
+import { Person } from '../types';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams<{ slug: string }>();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const searchParams = new URLSearchParams(location.search);
+  const selectedCentury = searchParams.get('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const query = searchParams.get('query');
+  const sex = searchParams.get('sex');
+
+  const filterPeople = (persons: Person[]) => {
+    return persons.filter(person => {
+      if (
+        selectedCentury &&
+        Math.floor(person.born / 100) + 1 !== parseInt(selectedCentury)
+      ) {
+        return false;
+      }
+
+      if (query && !person.name.toLowerCase().includes(query.toLowerCase())) {
+        return false;
+      }
+
+      if (sex && person.sex !== sex) {
+        return false;
+      }
+
+      return true;
+    });
+  };
+
+  const sortPeople = (
+    filteredPeople: Person[],
+    sortField: string,
+    sortOrder: string,
+  ) => {
+    return people.sort((a, b) => {
+      const aValue = a[sortField as keyof Person];
+      const bValue = b[sortField as keyof Person];
+
+      if (aValue < bValue) {
+        return sortOrder === 'asc' ? -1 : 1;
+      }
+
+      if (aValue > bValue) {
+        return sortOrder === 'asc' ? 1 : -1;
+      }
+
+      return 0;
+    });
+  };
+
+  const findSlug = (list: Person[], name: string) => {
+    const person = list.find(pers => pers.name === name);
+
+    return person && person.slug;
+  };
+
+  const updateSearchParams = (newParams: Record<string, string>) => {
+    const newSearchParams = new URLSearchParams(location.search);
+
+    Object.entries(newParams).forEach(([key, value]) => {
+      newSearchParams.set(key, value);
+    });
+    navigate(`${location.pathname}?${newSearchParams.toString()}`, {
+      replace: true,
+    });
+  };
+
+  const handleSort = (field: string) => {
+    let newOrder = 'asc';
+
+    if (sort === field && order === 'asc') {
+      newOrder = 'desc';
+    }
+
+    if (sort === field && order === 'desc') {
+      newOrder = '';
```

Comment:
When the sort order is set to an empty string, it might be more intuitive to remove the `sort` and `order` parameters from the URL instead of setting them to empty values. This can be done by updating the `updateSearchParams` function to handle empty values appropriately.